src/data-workspace/data-details-sidebar/comment.js

```diff
@@ -1,65 +1,49 @@
-import { useDataMutation, useAlert } from '@dhis2/app-runtime'
 import i18n from '@dhis2/d2-i18n'
-import {
-    CircularLoader,
-    Button,
-    ButtonStrip,
-    ReactFinalForm,
-    TextAreaFieldFF,
-} from '@dhis2/ui'
+import { CircularLoader, Button, ButtonStrip, TextAreaFieldFF } from '@dhis2/ui'
 import PropTypes from 'prop-types'
 import React, { useState } from 'react'
+import { Form, Field } from 'react-final-form'
+import { useContextSelection } from '../../context-selection/index.js'
 import { ExpandableUnit } from '../../shared/index.js'
+import { useSetDataValueCommentMutation } from '../use-data-value-mutation/index.js'
 import styles from './comment.module.css'
 import LoadingError from './loading-error.js'
 
-// TODO
-const mutation = {}
+const title = i18n.t('Comment')
+const errorMessage = i18n.t(
+    'There was a problem loading the comment for this data item'
+)
 
-export default function CommentUnit({ comment }) {
-    const [editing, setEditing] = useState(false)
-
-    const errorAlert = useAlert(
-        i18n.t('There was a problem saving the data item comment.'),
-        { critical: true }
-    )
-
-    //
-    // eslint-disable-next-line no-unused-vars
-    const [updateComment, { loading, error }] = useDataMutation(mutation, {
-        onError: () => errorAlert.show(),
-    })
-
-    if (loading) {
-        return <CircularLoader small />
-    }
+function CommentEditForm({ item, open, setOpen, syncComment, onCancel }) {
```

Comment:
General performance comment while testing adding/editing comments.

1. The Comment component keeps re-rendering while typing (minor)
2. The whole form re-renders twice when the comment is saved (maybe this is tackled in the other performance fixes we're doing)

![chrome-capture-2022-6-27 (1)](https://user-images.githubusercontent.com/1014725/181296796-b34be6f4-104e-4566-b602-b9444dae467b.gif)

Reply:
Yes, this has been addressed by the performance fixes which have been merged yesterday